plugin/trino-redis/src/main/java/io/trino/plugin/redis/RedisRecordCursor.java

```diff
@@ -340,6 +362,43 @@ private ScanParams setScanParams()
         return null;
     }
 
+    private void setPushdownKeys()
+    {
+        String keyStringPrefix = redisJedisManager.getRedisConnectorConfig().isKeyPrefixSchemaTable()
+                ? scanParams.match().substring(0, scanParams.match().length() - 1)
+                : EMPTY_STRING;
+        TupleDomain<ColumnHandle> constraint = split.getConstraint();
+        Map<ColumnHandle, Domain> domains = constraint.getDomains().orElseThrow();
+
+        for (Map.Entry<ColumnHandle, Domain> entry : domains.entrySet()) {
+            if (((RedisColumnHandle) entry.getKey()).isKeyDecoder()) {
+                Domain domain = entry.getValue();
+                if (domain.isSingleValue()) {
+                    String value = ((Slice) domain.getSingleValue()).toStringUtf8();
+                    keys = keyStringPrefix.isEmpty() || value.contains(keyStringPrefix) ? Lists.newArrayList(value) : emptyList();
+                    log.debug("Set pushdown keys %s with single value", keys.toString());
+                    return;
+                }
+                else {
+                    ValueSet valueSet = domain.getValues();
+                    if (valueSet instanceof SortedRangeSet) {
+                        Ranges ranges = ((SortedRangeSet) valueSet).getRanges();
+                        List<Range> rangeList = ranges.getOrderedRanges();
+                        if (rangeList.stream().allMatch(Range::isSingleValue)) {
+                            keys = rangeList.stream()
+                                    .map(range -> ((Slice) range.getSingleValue()).toStringUtf8())
+                                    .filter(str -> keyStringPrefix.isEmpty() || str.contains(keyStringPrefix))
+                                    .collect(toList());
+                            log.debug("Set pushdown keys %s with sorted range values", keys.toString());
+                            return;
+                        }
+                    }
+                }
+            }
+        }
+        keys = new ArrayList<>();
```

Comment:
It would be nice not to depend on mutable list. No changes requested as pre-existing issue.